TestFoundation/TestNSDateFormatter.swift

```diff
@@ -277,5 +278,17 @@ class TestNSDateFormatter: XCTestCase {
         XCTAssertEqual(f.string(from: testDate), "11-03-2016")
         
     }
-    
+
+    func test_setLocalizedDateFormatFromTemplate() {
+        let locale = Locale(identifier: DEFAULT_LOCALE)
+        let template = "EEEE MMMM d y hhmmss a zzzz"
+
+        let f = DateFormatter()
+        f.locale = locale
+        f.setLocalizedDateFormatFromTemplate(template)
```

Comment:
I am one of the Foundation "guys" 😉.  Looking at the implementation of the obj-c version of this method, the method essentially is meant to be purely a convenience so what you've implemented here is fine.